### PR TITLE
Copyediting of descriptions for Advanced Node & Express challenges

### PR DESCRIPTION
**Pull request by mshecket**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #XXXXX

I did some copyediting of the Advanced Node and Express challenges in English for grammar and style. I didn't touch the tests. There were some sentences I didn't understand, so I left them alone. My goal was to increase the clarity of the descriptions without changing the meaning of any of them. If folks are satisfied with what I did here, I'll plan to take a look at the descriptions of other challenges to see if I can suggest any improvements.

### CODE REVIEW

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Thanks for the work you put into this. I am happy with most of these changes.

Preferably, I would like to see the [contributors docs](https://contribute.freecodecamp.org/#/how-to-work-on-coding-challenges?id=formatting-challenge-text) have a dedicated section on the styling for when to use `code` tags, and when to use _em_ tags, etc. EDIT: I see there is, but it needs to be updated/extended.

I tried to not go _comma crazy_ on the sentences, but here we are.

**NOTE:** If you do go ahead an commit these suggestions, remember that you can use the _batch suggestion commit_ feature in GitHub (in the Files Changed tab). Just to prevent a dozen commits

**Comment by mshecket** [comment]
Not gonna lie--I must confess I am a comma fan. However, I have no need to die on Comma Hill.

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implement-the-serialization-of-a-passport-user.english.md
Comment (mshecket) [reply curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implement-the-serialization-of-a-passport-user.english.md:17]: There's another error here I missed...
```suggestion
Now we want to connect to our database then start listening for requests. The purpose of this is to not allow requests before our database is connected or if there is a database error. To accomplish this, you will want to encompass your serialization and your app routes in the following code:
```

**Comment by mshecket** [comment]
> the batch suggestion commit feature
Oops...sorry for the noob move. Next time I'll know.

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/authentication-strategies.english.md
Comment (Sky020) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/authentication-strategies.english.md:15]:
```suggestion
Now you will have to tell passport to <b>use</b> an instantiated <code>LocalStrategy</code> object with a few settings defined. Make sure this (as well as everything from this point on) is encapsulated in the database connection since it relies on it!
```

**Review by ShaunSHamilton — DISMISSED** [review]
As I am going through this section more and more, I am wanting to change more and more.

Unless another contributor disagrees, I suggest we hold off on this PR, for now. I am going to open an issue discussing the format we should stick to.

**Review by ShaunSHamilton — DISMISSED** [review]
Thank you, for getting this out. These changes LGTM 👍 

Just for future changes like this, it might be better to open seperate PRs for each section (eg _Advanced Node and Express_ changes in one PR, _Quality Assurance and Testing with Chai_ changes in another)

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/clean-up-your-project-with-modules.english.md
Comment (Manish-Giri) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/clean-up-your-project-with-modules.english.md:11]:
```suggestion
Right now, everything you have is in your <code>server.js</code> file. This can lead to hard to manage code that isn't very expandable.
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/clean-up-your-project-with-modules.english.md
Comment (Manish-Giri) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/clean-up-your-project-with-modules.english.md:25]:
```suggestion
Finally, take all of the routes in your server and paste them into your new files, and remove them from your server file. Also take the <code>ensureAuthenticated</code> function, since it was specifically created for routing. Now, you will have to correctly add the dependencies in which are used, such as <code>const passport = require('passport');</code>, at the very top, above the export line in your <code>routes.js</code> file.
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/how-to-use-passport-strategies.english.md
Comment (Manish-Giri) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/how-to-use-passport-strategies.english.md:11]:
```suggestion
In the <code>index.pug</code> file supplied, there is actually a login form. It has previously been hidden because of the inline JavaScript <code>if showLogin</code> with the form indented after it. Before <code>showLogin</code> as a variable was never defined, so it never rendered the code block containing the form. Go ahead and on the <code>res.render</code> for that page add a new variable to the object <code>showLogin: true</code>. When you refresh your page, you should then see the form! This form is set up to <b>POST</b> on <em>/login</em>, so this is where we should set up to accept the POST and authenticate the user.
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implementation-of-social-authentication.english.md
Comment (Manish-Giri) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implementation-of-social-authentication.english.md:15]:
```suggestion
Obtaining your <em>Client ID and Secret</em> from GitHub is done in your account profile settings under 'developer settings', then '<a href='https://github.com/settings/developers' target='_blank'>OAuth applications</a>'. Click 'Register a new application', name your app, paste in the url to your Repl.it homepage (<b>Not the project code's url</b>), and lastly, for the callback url, paste in the same url as the homepage but with <code>/auth/github/callback</code> added on. This is where users will be redirected for us to handle after authenticating on GitHub. Save the returned information as <em>'GITHUB_CLIENT_ID'</em> and <em>'GITHUB_CLIENT_SECRET'</em> in your <code>.env</code> file.
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/registration-of-new-users.english.md
Comment (Manish-Giri) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/registration-of-new-users.english.md:11]:
```suggestion
Now we need to allow a new user on our site to register an account. On the <code>res.render</code> for the home page add a new variable to the object passed along--<code>showRegistration: true</code>. When you refresh your page, you should then see the registration form that was already created in your <code>index.pug</code> file! This form is set up to <b>POST</b> on <code>/register</code>, so this is where we should set up to accept the <b>POST</b> and create the user object in the database.
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/send-and-display-chat-messages.english.md
Comment (Manish-Giri) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/send-and-display-chat-messages.english.md:12]:
```suggestion
It's time you start allowing clients to send a chat message to the server to emit to all the clients! In your <code>client.js</code> file, you should see there is already a block of code handling when the message form is submitted.
```

**Review by Manish-Giri — DISMISSED** [review]
Probably best to adhere to one convention around file name formatting.

**Review by raisedadead — DISMISSED** [review]

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/authentication-strategies.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/authentication-strategies.english.md:31]:
```suggestion
This is defining the process to use when we try to authenticate someone locally. First, it tries to find a user in our database with the username entered, then it checks for the password to match, then finally, if no errors have popped up that we checked for, like an incorrect password, the <code>user</code>'s object is returned and they are authenticated.
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
Please add code tags.

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/create-new-middleware.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/create-new-middleware.english.md:11]:
```suggestion
As is, any user can just go to <code>/profile</code> whether they have authenticated or not, by typing in the url. We want to prevent this, by checking if the user is authenticated first before rendering the profile page. This is the perfect example of when to create a middleware.
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/how-to-use-passport-strategies.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/how-to-use-passport-strategies.english.md:13]:
```suggestion
For this challenge you should add the route <code>/login</code> to accept a POST request. To authenticate on this route, you need to add a middleware to do so before then sending a response. This is done by just passing another argument with the middleware before your <code>function(req,res)</code> with your response! The middleware to use is <code>passport.authenticate('local')</code>.
```

File: curriculum/challenges/english/06-quality-assurance/quality-assurance-and-testing-with-chai/run-functional-tests-on-api-endpoints-using-chai-http.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/06-quality-assurance/quality-assurance-and-testing-with-chai/run-functional-tests-on-api-endpoints-using-chai-http.english.md:17]:
```suggestion
Don't send a name in the query; the endpoint responds with <code>hello Guest</code>.
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/serialization-of-a-user-object.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/serialization-of-a-user-object.english.md:14]:
```suggestion
To set this up properly, we need to have a serialize function and a deserialize function. In Passport, we create these with <code>passport.serializeUser( OURFUNCTION )</code> and <code>passport.deserializeUser( OURFUNCTION )</code>
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implementation-of-social-authentication.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implementation-of-social-authentication.english.md:15]:
```suggestion
Obtaining your <em>Client ID and Secret</em> from GitHub is done in your account profile settings under 'developer settings', then '<a href='https://github.com/settings/developers' target='_blank'>OAuth applications</a>'. Click 'Register a new application', name your app, paste in the url to your Repl.it homepage (<b>Not the project code's url</b>), and lastly, for the callback url, paste in the same url as the homepage but with <code>/auth/github/callback</code> added on. This is where users will be redirected for us to handle after authenticating on GitHub. Save the returned information as <code>'GITHUB_CLIENT_ID'</code> and <code>'GITHUB_CLIENT_SECRET'</code> in your <code>.env</code> file.
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/authentication-strategies.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/authentication-strategies.english.md:13]:
```suggestion
Add <code>passport-local</code> as a dependency and add it to your server as follows: <code>const LocalStrategy = require('passport-local');</code>
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/communicate-by-emitting.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/communicate-by-emitting.english.md:32]:
```suggestion
Now, you can implement a way for your client to listen for this event! Similar to listening for a connection on the server, you will use the <code>on</code> keyword.
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/how-to-use-passport-strategies.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/how-to-use-passport-strategies.english.md:11]:
```suggestion
In the <code>index.pug</code> file supplied, there is actually a login form. It has previously been hidden because of the inline JavaScript <code>if showLogin</code> with the form indented after it. Before <code>showLogin</code> as a variable was never defined, so it never rendered the code block containing the form. Go ahead and on the <code>res.render</code> for that page add a new variable to the object <code>showLogin: true</code>. When you refresh your page, you should then see the form! This form is set up to <b>POST</b> on <code>/login</code>, so this is where we should set up to accept the POST and authenticate the user.
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implement-the-serialization-of-a-passport-user.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implement-the-serialization-of-a-passport-user.english.md:13]:
```suggestion
To do this, add your database's connection string (for example: <code>mongodb+srv://:@cluster0-jvwxi.mongodb.net/?retryWrites=true&w=majority</code>) to the environment variable <code>MONGO_URI</code>. This is used in the <code>connection.js</code> file.
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/how-to-use-passport-strategies.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/how-to-use-passport-strategies.english.md:15]:
```suggestion
<code>passport.authenticate</code> can also take some options as an argument such as: <code>{ failureRedirect: '/' }</code> which is incredibly useful, so be sure to add that in as well. The response after using the middleware (which will only be called if the authentication middleware passes) should be to redirect the user to <code>/profile</code> and that route should render the view <code>profile.pug</code>.
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/how-to-use-passport-strategies.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/how-to-use-passport-strategies.english.md:17]:
```suggestion
If the authentication was successful, the user object will be saved in <code>req.user</code>.
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implementation-of-social-authentication-ii.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implementation-of-social-authentication-ii.english.md:13]:
```suggestion
To set up the GitHub strategy, you have to tell Passport to use an instantiated <code>GitHubStrategy</code>, which accepts 2 arguments: an object (containing <code>clientID</code>, <code>clientSecret</code>, and <code>callbackURL</code>) and a function to be called when a user is successfully authenticated, which will determine if the user is new and what fields to save initially in the user's database object. This is common across many strategies, but some may require more information as outlined in that specific strategy's GitHub README. For example, Google requires a <em>scope</em> as well which determines what kind of information your request is asking to be returned and asks the user to approve such access. The current strategy we are implementing has its usage outlined <a href='https://github.com/jaredhanson/passport-github/' target='_blank'>here</a>, but we're going through it all right here on freeCodeCamp!
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implementation-of-social-authentication.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implementation-of-social-authentication.english.md:13]:
```suggestion
Strategies with OAuth require you to have at least a <em>Client ID</em> and a <em>Client Secret</em> which is a way for the service to verify who the authentication request is coming from and if it is valid. These are obtained from the site you are trying to implement authentication with, such as GitHub, and are unique to your app--<b>THEY ARE NOT TO BE SHARED</b> and should never be uploaded to a public repository or written directly in your code. A common practice is to put them in your <code>.env</code> file and reference them like so: <code>process.env.GITHUB_CLIENT_ID</code>. For this challenge we're going to use the GitHub strategy.
```

**Review by RandellDawson — APPROVED** [review]
I changed several more `<em>` tags to `<code>` and now it looks good.

**Comment by RandellDawson** [comment]
Congrats @mshecket on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 📝

@Sky020 I was originally going to merge after committing @Manish-Giri's comments, but then decided to go ahead and change any `em` to `code` that showed up in the diff for the files.  There may be more, but we needed to get this PR merged today since it touches so many files, before we merge #39024

**Comment by Sky020** [comment]
@RandellDawson Absolutely. I was expecting this to be merged, after _Manish-Giri's_ changes. 

The only thing I would want changed is the list in `social-authentication`, because it affects the font. But that will be a separate PR.

**Pull request merged**
